opm/autodiff/StandardWell_impl.hpp

```diff
@@ -506,6 +516,29 @@ namespace Opm
             for (int componentIdx = 0; componentIdx < num_components_; ++componentIdx) {
                 cq_s[componentIdx] = cmix_s[componentIdx] * cqt_is; // * b_perfcells_dense[phase];
             }
+
+            // calculating the perforation solution gas rate and solution oil rates
+            if (well_type_ == PRODUCER) {
+                if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx) && FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
+                    const unsigned oilCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::oilCompIdx);
+                    const unsigned gasCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::gasCompIdx);
+                    // TODO: the formulations here remain to be tested with cases with strong crossflow through production wells
+                    // s means standard condition, r means reservoir condition
+                    // q_os = q_or * b_o + rv * q_gr * b_g
+                    // q_gs = q_gr * g_g + rs * q_or * b_o
+                    // d = 1.0 - rs * rv
+                    // q_or = 1 / (b_o * d) * (q_os - rv * q_gs)
+                    // q_gr = 1 / (b_g * d) * (q_gs - rs * q_os)
+
```

Comment:
With the discussion introduced with https://github.com/OPM/opm-simulators/pull/1441 , we also need to consider the the under-saturated situation here. And also, the sources for employed `rs` and `rv` also need to be discussed.